### DISCUSSION TITLE
EISPP 2020 - Full 2015 Video Series with 2020 Embellishments

### DISCUSSION THREAD

**Discussion by bshambaugh** [bshambaugh/eispp#24] [Ideas]
**Original:** 
seeAlso : [EISPP 2015 Wireframes](http://bshambaugh.org/eispp/)
seeAlso:  [YouTube 2015 Wireframe Animations](https://www.youtube.com/playlist?list=PLbVZNfQhcZ3eG_nbgKbC1KKtMXlIjnEsd)

**Comment by bshambaugh** [comment]
**Stuff learned 2020 and beyond:** (posted in FISSION Discord lobby on 5/30/2022)

5/30/2022
Meanwhile...

tl;dr: oh cool. someone with astounding skill works with functional programming in the web3 space. Sitting in on some w3c ccg calls and using web search tells me this ...maybe there is time for discussion at some point in the future. ... I know I need to learn this: http://learnyouahaskell.com/ ..and many many other things ..I was just trying to keep up when I met Marko and Ryan ....or some more involving FP, it might help me while also looking at A.C.T..... but there really isn't an off position to curiosity
despite present knowledge:
In late 2019/ early 2020 I came across http://www.mm-adt.org/ . https://markorodriguez.com/ was doing most of the programming and https://www.wisnesky.net/ was drawing on his encyclopaedia of math knowledge. I spent quite a bit of time
in the development channels and got to know Marko and Ryan, Around mid-way through I discovered https://ceramic.network/, and I knew through my exposure to the semantic web that data management could be a problem. Calbimonte was a paper that
I turned up years ago that involved sensor data management: http://ceur-ws.org/Vol-839/calbimonte.pdf .
Due to Ryan, I discovered Dr. Spivak. Dr. Spivak's talk on categorical data bases  (https://www.youtube.com/watch?v=bk36__qkhrk )resonated with me because I knew expecting an upper ontology to somehow make all data interoperable would be irrational. Rather, mapping
would be in order. Watching the ceramic channel (IIRC) led to discovery of https://github.com/inkandswitch/cambria-project. This seems great and might be something that would be faster than learning the applied math and developing myself. It is likely more than I could do with my early knowledge.
I did ask questions anyways during my journey. I found these links useful: json-schema.org/, https://dashjoin.medium.com/json-schema-schema-org-json-ld-whats-the-difference-e30d7315686a, https://depositonce.tu-berlin.de/handle/11303/2617 . The last is a dissertation
by Benjamin Braatz that Ryan Wisnesky provided me with when I asked about Applied Category Theory and RDF. Examining the problem myself has been something I've been wanting to dive deeper into. Along the way I bought Bartosz Milewski's Piggy book and several
other books that Marko bought.
mm-ADT
[mm-ADT](http://www.mm-adt.org/)
A Multi-Model Abstract Data Type
Marko A. Rodriguez
[Marko A. Rodriguez](https://markorodriguez.com/)
TMI: sometimes my mind gets in a log jam and have to clear it out. I don't expect a detailed analysis of this.
bshambaugh
 — 
05/30/2022
https://layeredschemas.org/ is a great thing I discovered at IIW.  Burak Serdar is awesome. 
[Layered Schema Architecture](https://layeredschemas.org/)
Layered schemas data capture, semantic interoperability, knowledge graphs
"Layered Schema Architecture (LSA) enables semantic interoperability between heterogeneous systems. "
I will need to come back to this: https://subconscious.substack.com/p/layered-protocols?s=r
[Layered protocols](https://subconscious.substack.com/p/layered-protocols?s=r)
Layering is a special form of modularity that restricts interactions between modules. A module assigned to a layer can use any of the modules in the same or in a lower layer; however, it cannot use a module belonging to a higher layer.

**Comment by bshambaugh** [comment]
6/06/2022
Ryan Wisnesky will be having a meetup tomorrow night (online). He works with Dr. Spivak. He's always a hoot. https://www.meetup.com/Category-Theory/ 

https://fission.codes/blog/project-cambria-overview/ (shared by Boris)

6/14/2022
Thanks. to me the semantic web is something to learn from to me to inform thinking in json-schema, or richer data if chosen. it's just a graph. https://core.ac.uk/display/57704480 was just prior art I turned up to inform further exploration. I'll take a closer look at the link you gave. thank you.

**Comment by bshambaugh** [comment]
(6/14/2022)
I don't know where to put this, but I had the idea I might talk to ..... at some point. It's not well formed, but at least I said it. ""In the future, speaking of interop, I'm interested in how data stored in the ecosystem will play together. I think that the semantic web is a lesson to learn from.
I agree with Dr. Spivak that with data (and I say by extension distributed systems) you don't want a centralized source of meaning like an upper ontology.
Instead, focus on rigorously mapping.

I have some links. ...

https://core.ac.uk/display/57704480 (Formal Modelling and Application of Graph Transformations in the Resource Description Framework)
https://github.com/inkandswitch/cambria-project (Schema evolution with bi-directional lenses.)
https://github.com/cloudprivacylabs/lsa (layered schema architecture)
https://dashjoin.medium.com/json-schema-schema-org-json-ld-whats-the-difference-e30d7315686a "" I shared this in the TBD discord when Gabe shared https://github.com/decentralized-identity/schema-directory .  oh, btw I know Ryan Wisnesky who works with Dr. Spivak. https://www.wisnesky.net/ on https://www.categoricaldata.net/ (applied category theory for databases). I'm a student not an expert...and really a lot on my plate atm...but this is definitely a TODO for whoever is interested.
